### ISSUE TITLE
Text is cut off in SVG exported from live editor

### ISSUE THREAD

**Issue by p3k** [mermaid-js/mermaid#790]
The following Mermaid markup generates this [live view](https://mermaidjs.github.io/mermaid-live-editor/#/view/eyJjb2RlIjoiZ3JhcGggVERcbkFbU29tZXRoaW5nXSAtLT58YmVnZXRzfCBCW1NvbWV0aGluZ10iLCJtZXJtYWlkIjp7fX0):

```mermaid
graph TD
A[Something] -->|begets| B[Something]
```
(Config is `{}`.)

![image](https://user-images.githubusercontent.com/16631/52178038-87b1d880-27c9-11e9-981c-d6bb8a927be0.png)

However, in the [exported SVG](https://github.com/knsv/mermaid/files/2825607/mermaid-diagram-20190203152903.svg.txt) all text boxes are cropped on the right side:

![image](https://user-images.githubusercontent.com/16631/52178048-b7f97700-27c9-11e9-8aed-00eeadf4028e.png)

Am I missing something?

**Comment by CalvinRodo** [comment]
I'm running into this same issue currently

**Comment by jcpst** [comment]
I also run into this same issue when using the library in a web application, where config is `{}`.

**Comment by Uinelj** [comment]
Issue present in the generated SVG from the live demo too (tested on Chrome, Firefox):

![image](https://user-images.githubusercontent.com/4410814/53037162-c5755900-3479-11e9-92f7-0ec9ed6737d6.png)

**Comment by hexblot** [comment]
In some cases text is cropped even in the live editor, see [a very basic example here](https://mermaidjs.github.io/mermaid-live-editor/#/view/eyJjb2RlIjoiZ3JhcGggVERcbkFbQ2hyaXN0bWFzXSAtLT58MTkyLjE2OC4xLjEwMHwgQihHbyBzaG9wcGluZykiLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ)

Exporting to SVG makes the issue more pronounced.

**Comment by trueskawka** [comment]
I was able to address this issue when using the CLI by adding CSS to the parent element for `edgeLabel`. The `foreignObject` width is too narrow for labels to be properly displayed, so it should allow for overflow (I think the math might be slightly off when calculating the label width).

```
.label foreignObject {
    overflow: visible;
}
```

**Comment by FiyaFly** [comment]
> I was able to address this issue when using the CLI to generate PNG by adding CSS to the parent element for `edgeLabel`. The `foreignObject` width is too narrow for labels to be properly displayed, so it should allow for overflow (I think the math might be slightly off when calculating the label width).
> 
> ```
> .label foreignObject {
>     overflow: visible;
> }
> ```
> 
> It doesn't seem to be working for SVG, but it might be a similar issue.
> 

To add to this, in the SVG every class is marked with an ID, which is noted at the top, e.g. ```svg id="mermaid-1556643060555"```. Also, if you add ```font-size: 90%;``` as well, it looks almost perfect (at least on my screen, haven't tested elsewhere). Which would make the css similar to this, using the svg line as an example:

```css
#mermaid-1556643060555 .label foreignObject {
    font-size: 90%;
    overflow: visible;
}
```

I stumbled across this issue trying to figure out why my SVG's were coming out poor, but I've only been using mermaid for a day lol. Hopefully this helps others, and eventually helps to fix it in the editor/mermaid itself. I don't know about using this via CLI. Still working on getting the CLI working for me at all lol.

**Comment by happycollision** [comment]
Well, it's not stale! ;-)

**Comment by Drunkman** [comment]
I add this to config, and it work well. 
```json
{
    "themeCSS": ".label foreignObject { font-size: 14px; }"
}
```
But I still don't know why mermaid add a style that make the font-size large so that graph label be cut off.
![image](https://user-images.githubusercontent.com/10599065/62998885-6aa84180-be9f-11e9-881c-9e7d3cafb186.png)

**Comment by Dunning-Kruger** [comment]
This does not seem to be happening in the current version: 8.3.1
[Link to live editor.](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbkFbU29tZXRoaW5nXSAtLT58YmVnZXRzfCBCW1NvbWV0aGluZ10iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ)

**Comment by pdxjohnny** [comment]
I still get clipping on the export

![image](https://user-images.githubusercontent.com/5950433/65647148-9a745a00-dfb2-11e9-8204-7e36f2b2f5ef.png)

**Comment by Dunning-Kruger** [comment]
Note that right now the project has a lot more requests piled up than it can handle.
[We would be happy to accept new collaborators! ](https://github.com/knsv/mermaid/issues/866)

**Comment by GDFaber** [comment]
This issue seems to be solved in mermaid 8.4.4 as the svg download of the new live editor does not show the problem anymore:

https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbkFbU29tZXRoaW5nXSAtLT58YmVnZXRzfCBCW1NvbWV0aGluZ10iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ==

**Comment by p3k** [comment]
confirmed! great, thanks a lot. closing this issue.
